scripts/devel/build_and_install.py

```diff
@@ -40,10 +42,25 @@ def call_abi_checker(workspace_root, ros_version, env):
     pkg_names = [pkg.name for pkg in pkgs.values()]
     assert pkg_names, 'No packages found in the workspace'
 
+    # Filter packages in source space that has been released
+    index_file = rosdistro.get_index(rosdistro.get_index_url())
+    distro_file = rosdistro.get_cached_distribution(index_file, env['ROS_DISTRO'])
+    pkg_names_released = []
+    for pkg_name in pkg_names:
+        try:
+            if distro_file.get_release_package_xml(pkg_name):
```

Comment:
Why does it need to fetch the `package.xml` content?

Reply:
It does not need to fetch the package xml. It does need to check if a given package name has been released. What would be the best way of doing this?

Reply:
A subset of what the function is doing internally (and you don't need a full `Distribution` for that anymore but only a `DistributionFile` so the comment above might not apply anymore): https://github.com/ros-infrastructure/rosdistro/blob/1bd3c008404d54df0299361afe5a8a4ebf3e3da7/src/rosdistro/distribution.py#L66-L73

Reply:
Thanks for the explanation. I simplified the logic in 9f96596 to just get the distribution file and try to access to the package using `dist_file.release_packages[pkg_name]`. Good enough?